state_version_integration_test.go

```diff
@@ -333,6 +333,12 @@ func TestStateVersionsRead(t *testing.T) {
 		sv.Outputs = nil
 
 		assert.Equal(t, svTest, sv)
+		assert.NotEmpty(t, svTest, svTest.ResourcesProcessed)
```

Comment:
This is what I meant! Thanks.

I am slightly worried that some of these will indeed be empty if the backend hasn't yet processed the state. The way to fix this would be to retry while ResourcesProcessed was false. Or we could just wait and see if it flakes!

Reply:
@brandonc Wouldn't it stand true that if ResourcesProcessed were false, it would still validate on NotEmpty? Like regardless of which bool, it will still always contain a value?

Reply:
Ah, the attr returning `false` could be an indicator that the other exposed attributes may be empty. Got it.

Reply:
`false` is actually empty as well because it's the default value of a boolean. This is the source of a lot of bugs, I think! Check this out:

https://go.dev/play/p/Qsz5Q7FdAWU

Reply:
Could this example in the [State Version API docs](https://www.terraform.io/cloud-docs/api-docs/state-versions#list-state-versions-for-a-workspace) be incorrect? The response has `resources-processed: false` but the other attributes I exposed are populated. 

```
"modules": {
          "root": {
              "null-resource": 1,
               "data.terraform-remote-state": 1
                }
            },
            "providers": {
                "provider[\"terraform.io/builtin/terraform\"]": {
                    "data.terraform-remote-state": 1
                },
                "provider[\"registry.terraform.io/hashicorp/null\"]": {
                    "null-resource": 1
                }
            },
            "resources": [
                {
                    "name": "other_username",
                    "type": "data.terraform_remote_state",
                    "count": 1,
                    "module": "root",
                    "provider": "provider[\"terraform.io/builtin/terraform\"]"
                },
                {
                    "name": "random",
                    "type": "null_resource",
                    "count": 1,
                    "module": "root",
                    "provider": "provider[\"registry.terraform.io/hashicorp/null\"]"
                }
            ],
            "resources-processed": false,
```

Reply:
Yes, it is. If resources_processed is false, you would expect the resources key to be empty.

Reply:
Ah, ok, according to the [docs](https://pkg.go.dev/github.com/realistschuckle/testify/assert?utm_source=godoc#NotEmpty), there's a `func NotEmpty` workaround. 
```
Empty asserts that the specified object is NOT empty. I.e. not nil, "", false, 0 or a slice with len == 0. 
Returns whether the assertion was successful (true) or not (false).
```


```
if assert.NotEmpty(t, obj) {
  assert.Equal(t, "two", obj[1])
}
```

So maybe `assert.Equal()` that the value is either `true` or `false`, using both `assert.Empty` and `assert.NotEmpty`. What do you think?